app/models/tag.rb

```diff
@@ -54,8 +53,17 @@ def taggings_count_cache_key
   end
 
   def write_taggings_to_redis(value)
+    # Atomically set the value while extracting the old value.
+    old_redis_value = REDIS_GENERAL.getset("tag_update_#{id}_value", value)
+
+    # If the value hasn't changed from the saved version or the REDIS version,
+    # there's no need to write an update to the database, so let's just bail
+    # out.
+    return value if value == old_redis_value && value == taggings_count_cache
+
+    # If we've reached here, then the value has changed, and we need to make
+    # sure that the new value is written to the database.
     REDIS_GENERAL.sadd("tag_update", id)
```

Comment:
We could make a change to the rake that deletes 0 use tags so that any tag which was not deleted and had uses had the count set correctly.

https://github.com/otwcode/otwarchive/blob/master/lib/tasks/tag_tasks.rake#L41

if this was t.taggings_count.zero? rather than t.taggings.count.zero?